ember-core/src/main/scala/org/http4s/ember/core/ChunkedEncoding.scala

```diff
@@ -29,6 +29,7 @@ private[ember] object ChunkedEncoding {
   def decode[F[_]](
       head: Array[Byte],
       read: F[Option[Chunk[Byte]]],
+      maxHeaderSize: Int,
```

Comment:
I think we shouldn't conflate `maxHeaderSize` (the max bytes of the trailer header section) and with what `maxChunkHeaderSize` is (the max size for chunked bodies?). I'm thinking we should probably bubble the latter out into a separate parameter. Maybe even a separate one for trailers?

Reply:
A separate one for trailers sounds good.  I'm still confused by maxChunkHeaderSize.

Reply:
So it looks like `maxChunkHeaderSize` is the max size of the chunk header for each chunk. so `4\r\n` has a length of 3.  Is it worth having a separate configuration parameter for that? Right now it's set to `maxHeaderSize`

Reply:
I'm going to hold off on adding these for now... We're slowly reaching the limit of class parameters we can take (we're at like 15 or 16 right now) so I want to group things together a bit to make it nicer